test/integration/ovs/openflow_test_utils.go

```diff
@@ -53,19 +53,21 @@ type ExpectFlow struct {
 
 func CheckFlowExists(t *testing.T, ovsCtlClient ovsctl.OVSCtlClient, tableID uint8, exist bool, flows []*ExpectFlow) []string {
 	flowList, _ := OfctlDumpTableFlows(ovsCtlClient, tableID)
+
+	install := "uninstall"
 	if exist {
-		for _, flow := range flows {
-			if !OfctlFlowMatch(flowList, tableID, flow) {
-				t.Errorf("Failed to install flow:\n%v\nExisting flows:\n%v", flow, flowList)
-			}
-		}
-	} else {
-		for _, flow := range flows {
-			if OfctlFlowMatch(flowList, tableID, flow) {
-				t.Errorf("Failed to uninstall flow:\n%v\nExisting flows:\n%v", flow, flowList)
-			}
+		install = "install"
+	}
+	failExist := false
+	for _, flow := range flows {
+		if (exist && !OfctlFlowMatch(flowList, tableID, flow)) || (!exist && OfctlFlowMatch(flowList, tableID, flow)) {
+			failExist = true
+			t.Errorf("Failed to %s flow:\n%v", install, flow)
 		}
 	}
+	if failExist {
```

Comment:
The bool is not needed
```suggestion
	if t.Failed() {
```

Reply:
Updated.

Reply:
Updated.